packages/Button/src/index.tsx

```diff
@@ -1,32 +1,34 @@
-/* eslint-disable react/button-has-type */
-import React from "react";
+import withUtils from "@blaze-react/utils";
+import React, { ButtonHTMLAttributes, FunctionComponent } from "react";
 
-interface IButtonProps {
+interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   disabled?: boolean;
-  submit?: boolean;
   children?: JSX.Element | string;
-  modifiers?: string;
-  attrs?: any;
+  type?: "button" | "submit" | "reset";
+  modifiers: string;
+  utils: {
+    classNames: (className: string | object, classNames?: object) => string;
+  };
 }
 
-const Button: React.SFC<IButtonProps> = ({
+const Button: FunctionComponent<IButtonProps> = ({
   disabled,
-  submit,
+  type,
   children,
   modifiers,
+  utils: { classNames },
   ...attrs
 }): JSX.Element => {
-  const type = submit ? "submit" : "button";
-  const allModifiers =
-    modifiers &&
-    modifiers
+  const buttonClassNames: string = classNames("button", {
+    [modifiers
```

Comment:
this modifiers.split.map etc should be in a constant and use constant there instead all this piece of code. Will be more readable